src/main/java/com/google/sps/data/Availability.java

```diff
@@ -12,39 +12,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package com.google.sps;
+package com.google.sps.data;
 
+import com.google.auto.value.AutoValue;
 import java.time.LocalDate;
 
 /**
  * Availability is a time range when a given person is available to offer interviews as an
- * Interviewer.
+ * interviewer.
  */
-public class Availability {
-  private String personEmail;
-  private TimeRange when;
-  private LocalDate date;
-
-  /** This constructor creates a new Availability object */
-  public Availability(String personEmail, TimeRange when, LocalDate date) {
-    this.personEmail = personEmail;
-    this.when = when;
-    this.date = date;
-  }
+@AutoValue
+public abstract class Availability {
+  abstract String email();
 
-  public void updateTimeRange(Timerange range) {
-    this.when = range;
-  }
+  abstract TimeRange when();
```

Comment:
These might need to be public so that they can be accessed outside of the `data` package.

`public abstract`.